lib/web_ui/lib/src/engine/text_editing/text_editing.dart

```diff
@@ -1361,6 +1361,7 @@ abstract class DefaultTextEditingStrategy with CompositionAwareMixin implements
   @override
   void setEditingState(EditingState? editingState) {
     lastEditingState = editingState;
+    _editingDeltaState = null;
```

Comment:
Nit: I think we can just put this in `handleChange`. And instead of just flushing the `editingDeltaState` when the `newEditingState != lastEditingState`, we should always flush it so we are ready for the next delta inference.